src/components/forms/forms.js

```diff
@@ -13,80 +21,122 @@ import {
+  'name',
+  'disabled',
+  'description',
+  'title',
+  'ariaLevel',
+  'defaultValue',
+  'hideTitle',
+  'style',
+  'validators',
+  'onChange',
+  'type'
+];
+
+// ----------------------------------------------- //
+// ======       MAIN FORM FIELD TYPES       ====== //
+// ----------------------------------------------- //
 export const FormFieldTypes = {
-  MULTITEXT: { defaultValue: [], component: formInputMultiText },
-  SLIDER: { defaultValue: false, component: formInputSlider },
-  RADIO: { defaultValue: null, component: formInputRadioGroup },
-  TEXT: { defaultValue: '', component: formInputGeneric },
-  NUMBER: { defaultValue: '', component: formInputGeneric },
+  MULTITEXT: {
+    defaultValue: [],
+    component: formInputMultiText,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'ariaDescribedby',
+      'inputStyle'
+    ],
+  },
+  SLIDER: {
+    defaultValue: false,
+    component: formInputSlider,
+    requiredProps: [],
+    optionalProps: [
+      'toggleText',
+    ],
+  },
+  RADIO: {
+    defaultValue: null,
+    component: formInputRadioGroup,
+    requiredProps: [
+      'options',
+      // 'options' example:
+      // [
+      //  {name: 'opt_1', text: 'Option 1'},
+      //  {name: 'other', text: 'Other', type: 'string'} <-- will have textbox
+      // ]
+    ],
+    optionalProps: [
+      'ariaDescribedBy',
+    ],
+    customPropValidation: customRadioPropValidation,
+  },
+  TEXT: {
+    defaultValue: '',
+    component: formInputGeneric,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+      'ariaDescribedby',
+    ],
+  },
+  NUMBER: {
+    defaultValue: '',
+    component: formInputGeneric,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+      'ariaDescribedby',
+    ],
+  },
   CHECKBOX: {
     defaultValue: (config) => (config?.valueType === 'string' ? '' : false),
-    component: formInputCheckbox
+    component: formInputCheckbox,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+    ],
   },
   SELECT: {
     defaultValue: (config) => (config?.isMulti ? [] : ''),
-    updateDefaultValue: ({ selectOptions, defaultValue }) => {
-      const isStringArr = isString(selectOptions[0]);
-      return isStringArr
-        ? { key: defaultValue, displayValue: defaultValue }
-        : defaultValue;
-    },
-    component: formInputSelect
+    updateDefaultValue: updateSelectDefaultValue,
+    component: formInputSelect,
+    requiredProps: [
+      'selectOptions'
+      // 'selectOptions' example:
+      // [
+      //  {displayText: 'Option 1'}
+      //   ^^^ can pass other fields in as extra info, e.g.:
+      //  {displayText: 'Dac 1', dacId: 213}
+      // ]
+    ],
+    optionalProps: [
+      'isCreatable', // allows user to input their own
+      'isMulti',
+      'placeholder',
+      'ariaDescribedby',
+      'required',
```

Comment:
Should this be a field? vs `FormValidators.REQUIRED`?

Reply:
agreed - i was just copying over what was there but this is a good opportunity to normalize these